Utilities/Mergers/test/test_ObjectStore.cxx

```diff
@@ -29,89 +30,114 @@ using namespace o2::framework;
-    obj->Fill(4);
+  DataRef ref = makeDataRef(obj);
 
-    DataRef ref = makeDataRef(obj);
+  auto objStore = object_store_helpers::extractObjectFrom(ref);
+  BOOST_CHECK(std::holds_alternative<TObjectPtr>(objStore));
 
-    auto objStore = object_store_helpers::extractObjectFrom(ref);
-    BOOST_CHECK(std::holds_alternative<TObjectPtr>(objStore));
+  auto objExtractedHisto = dynamic_cast<TH1I*>(std::get<TObjectPtr>(objStore).get());
+  BOOST_REQUIRE(objExtractedHisto != nullptr);
+  BOOST_CHECK_EQUAL(objExtractedHisto->GetEntries(), 1);
 
-    auto objExtractedHisto = dynamic_cast<TH1I*>(std::get<TObjectPtr>(objStore).get());
-    BOOST_REQUIRE(objExtractedHisto != nullptr);
-    BOOST_CHECK_EQUAL(objExtractedHisto->GetEntries(), 1);
+  delete ref.header;
+  delete ref.payload;
+  delete obj;
+}
 
-    delete ref.header;
-    delete ref.payload;
-    delete obj;
-  }
+BOOST_AUTO_TEST_CASE(TArrayOfHisto1D)
+{
+  TObjArray* array = new TObjArray();
+  array->SetOwner(true);
 
-  {
-    TObjArray* array = new TObjArray();
-    array->SetOwner(true);
+  TH1I* histo = new TH1I("histo 1d", "histo 1d", 100, 0, 100);
+  histo->Fill(5);
+  array->Add(histo);
 
-    TH1I* histo = new TH1I("histo 1d", "histo 1d", 100, 0, 100);
-    histo->Fill(5);
-    array->Add(histo);
+  DataRef ref = makeDataRef(array);
+  auto objStore = object_store_helpers::extractObjectFrom(ref);
+  BOOST_CHECK(std::holds_alternative<TObjectPtr>(objStore));
 
-    DataRef ref = makeDataRef(array);
-    auto objStore = object_store_helpers::extractObjectFrom(ref);
-    BOOST_CHECK(std::holds_alternative<TObjectPtr>(objStore));
+  auto objExtractedArray = dynamic_cast<TObjArray*>(std::get<TObjectPtr>(objStore).get());
+  BOOST_REQUIRE(objExtractedArray != nullptr);
+  BOOST_CHECK_EQUAL(objExtractedArray->GetEntries(), 1);
 
-    auto objExtractedArray = dynamic_cast<TObjArray*>(std::get<TObjectPtr>(objStore).get());
-    BOOST_REQUIRE(objExtractedArray != nullptr);
-    BOOST_CHECK_EQUAL(objExtractedArray->GetEntries(), 1);
+  delete ref.header;
+  delete ref.payload;
+  delete array;
+}
 
-    delete ref.header;
-    delete ref.payload;
-    delete array;
-  }
-}
\ No newline at end of file
+BOOST_AUTO_TEST_CASE(VectorOfHistos1D)
+{
+  auto histo = std::make_shared<TH1F>("histo 1d", "histo 1d", 100, 0, 100);
+  histo->Fill(5);
+  VectorOfTObjectPtr vectorWithData{histo};
+
+  auto vectorToDataRef = object_store_helpers::toRawPointers(vectorWithData);
+
+  DataRef ref = makeDataRef(&vectorToDataRef);
+  auto objStore = object_store_helpers::extractObjectFrom(ref);
+  BOOST_CHECK(std::holds_alternative<VectorOfTObjectPtr>(objStore));
+  auto extractedVector = std::get<VectorOfTObjectPtr>(objStore);
+  BOOST_CHECK(extractedVector.size() == 1);
+  auto* extractedHisto = dynamic_cast<TH1F*>(extractedVector[0].get());
+  BOOST_CHECK(gsl::span(histo->GetArray(), histo->GetSize()) == gsl::span(extractedHisto->GetArray(), extractedHisto->GetSize()));
+}
+
+BOOST_AUTO_TEST_SUITE_END()
```

Comment:
Thanks for refactoring the file to use SUITE and CASE. I was not aware of the possible distinction, indeed it is better this way. However, next time please make such improvements as separate commits or PRs, since it obfuscates the commit history.

Reply:
I thought that this refactoring of test suite, I was working on, was okay to be in this PR. But I will remember that for the future.